src/setting.rs

```diff
@@ -17,24 +17,38 @@
 //!
 //! The exact format still has to be reverse-engineered.
 
-use crate::util::nom_input_error_with_kind;
-use nom::error::ErrorKind;
-use nom::IResult;
+use binrw::{binrw, NullString};
 
-#[derive(Debug)]
+#[derive(Debug, PartialEq)]
+#[binrw]
+#[brw(little)]
 /// Represents a setting file.
 pub struct Setting {
     /// Size of the string data field (should be always 96).
+    #[br(assert(len_stringdata == 0x60))]
     pub len_stringdata: u32,
-    /// Name of the company ("PIONEER").
-    pub company: String,
+    /// Name of the brand.
+    ///
+    /// The value seems to depend on the kind of file:
+    ///
+    /// | File               | Value        |
+    /// | ------------------ | ------------ |
+    /// | `DEVSETTING.DAT`   | `PIONEER DJ` |
+    /// | `DJMMYSETTING.DAT` | `PioneerDJ`  |
+    /// | `MYSETTING.DAT`    | `PIONEER`    |
+    /// | `MYSETTING2.DAT`   | `PIONEER`    |
+    #[brw(pad_size_to = 0x20)]
+    pub brand: NullString,
     /// Name of the software ("rekordbox").
-    pub software: String,
+    #[brw(pad_size_to = 0x20)]
+    pub software: NullString,
     /// Some kind of version number.
-    pub version: String,
+    #[brw(pad_size_to = 0x20)]
+    pub version: NullString,
     /// Size of the `data` data in bytes.
     pub len_data: u32,
-    /// Unknown field.
+    /// The actual settings data.
+    #[br(args(len_data))]
```

Comment:
```suggestion
    #[brw(args(len_data))]
```

Reply:
Nope, that doesn't make sense. The length is needed for deserialization (to decide which data type is used), but when writing the byte size is determined by the data type. It doesn't make sense to pass the length data.

Reply:
you're right, it doesn't make much sense. Though I'm unsure whether it actually serializes it correctly in the current state, I'll have to check.

Reply:
It does, I added round-trip tests. But it's possible to input "wrong" values. Ideally, this field would be temp and automatically calculated.